fault/tester.py

```diff
@@ -53,3 +57,31 @@ def serialize(self):
     def compile_and_run(self, target="verilator", **kwargs):
         target_inst = self.make_target(target, **kwargs)
         target_inst.run()
+
+    def __check_interfaces_match(self, old_circuit, new_circuit):
+        old_port_names = old_circuit.interface.ports.keys()
+        for name in old_port_names:
+            if not hasattr(new_circuit, name):
+                raise ValueError(f"New circuit does not have port {name}")
+            old_kind = type(type(getattr(old_circuit, name)))
+            new_kind = type(type(getattr(new_circuit, name)))
+            if not (issubclass(new_kind, old_kind) or
+                    issubclass(old_kind, new_kind)):
+                raise ValueError("Types don't match")
+
+    def copy(self, new_circuit, clock=None):
+        """
+        Generates a copy of the tester for new_circuit
+        Checks that new_circut has exactly the same interface as self.circuit
+        """
+        self.__check_interfaces_match(self.circuit, new_circuit)
+        new_tester = Tester(new_circuit, clock)
+        for old_action in self.actions:
+            new_action = copy.copy(old_action)
```

Comment:
Maybe actions are required to implement a `retarget(new_circuit)` function? So we can just defer to those calls and don't need the `isinstance()` calls. Both are ok, the former seems cleaner to me -- also depends on if we expect that `retarget` function to be used anywhere else.